### ISSUE TITLE
Performance issues

### ISSUE THREAD

**Issue by lieblinger** [seaky/nspanel_pro_tools_apk#63]
Sorry for this topic cuase I know that this is not related to your tools but I'm facing huge performance issues with multiple NSPanel Pro, spend hours trying to improve the setup and really hope that you have some tips for me.

**What I've alread tried:**

- Improved the wifi signal, 
- debloated the panel, 
- used HA without any custom addons and
- also tried using very simple dashboards. 

Unfortunately without any luck.

Mainly I try to use FullyKiosk with Home Assistant but it's sooo slow it takes super long to load the dashboard and status updates takes about 10 seconds before a change appears. See the following video:

https://github.com/seaky/nspanel_pro_tools_apk/assets/10740088/9565e0b4-f10e-471e-8f41-8c2c8def5eb4

Is this normal or is there anything I could improve? I would be very thankful for any help!

**Comment by bobzer** [comment]
maybe you have too many apps running in the bckgrounnd ?
did you tried to debloat the apps ?
https://blakadder.com/nspanel-pro-secrets/

**Comment by lieblinger** [comment]
Thanks for your help! I already checked your guide and removed unnecessary apps. 

These are the apps running currently:
[activity.log](https://github.com/seaky/nspanel_pro_tools_apk/files/14300420/activity.log)

And these are all installed apps:
[packages.log](https://github.com/seaky/nspanel_pro_tools_apk/files/14300406/packages.log)

Is FKB and Home Assistant working responsive in your setup?

**Comment by bobzer** [comment]
i don't have FKB yet and i'm just testing in my desk so not much feedback but it seems much faster than yours

**Comment by seaky** [comment]
Hi,

I think its related to FK. I use the simple companion app, it loads about 10sec for the first time. After it is super responsive even so I use lovelance plugins. It is always on front and my app handles the sensors.

try the HA app...

**Comment by seaky** [comment]
Here is one of my device with fw1.5:

https://github.com/seaky/nspanel_pro_tools_apk/assets/5487684/a4656af6-b703-4fdf-887a-bcbb5e936d05

**Comment by seaky** [comment]
Check your home wifi my ping/down rate is about 3ms/100mbps

**Comment by kev007** [comment]
Check if your known_devices.yaml file is bloated. Mine was over 500,000 lines long and it absolutely killed my HA performance.

**Comment by lieblinger** [comment]
@seaky The performance in your video looks amazing. This is what I'd like to have. 

@kev007 Thanks for your tip. Unfortunately there's no known_devices.yaml on my installation.

I feel like my HA enviroment could be to heavy for this device. Cause on other devices like Samsung S4 TAB and a couple of RK3588 and RK3566 devices it works seamlessly. I found a supplier which offers these panels with an RK3566 chipset. Gonna try that.

Thanks again for all your help!

**Comment by lieblinger** [comment]
Just a little update. I've got a couple of RK3566 devices and they perform a way better. But I still had **huge** performance issues and checked again my plugins.

After I dived deep into troubleshooting performance with my installed plugins I recognized three which slowed down performance the most...

The worst was card_mod it doesnt matter if I add via resources in lovelace or frontend in configuration.yaml and button_card. Also the kiosk_mode plugin was slowing down everything a little bit. I had to move away from it and change to browser_mod.

Also Fully Kiosk Browser is slower than HA android app.

After these improvements everything seems to work better....
